packages/core/components/AnnotationPicker/index.tsx

```diff
@@ -27,31 +28,68 @@ interface Props {
  * downloading a manifest.
  */
 export default function AnnotationPicker(props: Props) {
-    const annotations = useSelector(metadata.selectors.getSortedAnnotations);
+    const annotations = useSelector(selection.selectors.getSortedAnnotations).filter(
+        (annotation) =>
+            !props.disabledTopLevelAnnotations ||
+            !TOP_LEVEL_FILE_ANNOTATION_NAMES.includes(annotation.name)
+    );
     const unavailableAnnotations = useSelector(
         selection.selectors.getUnavailableAnnotationsForHierarchy
     );
     const areAvailableAnnotationLoading = useSelector(
         selection.selectors.getAvailableAnnotationsForHierarchyLoading
     );
 
-    const items = annotations
-        .filter(
-            (annotation) =>
-                !props.disabledTopLevelAnnotations ||
-                !TOP_LEVEL_FILE_ANNOTATION_NAMES.includes(annotation.name)
-        )
-        .map((annotation) => ({
-            selected: props.selections.some((selected) => selected.name === annotation.name),
-            disabled:
-                !props.enableAllAnnotations &&
-                unavailableAnnotations.some((unavailable) => unavailable.name === annotation.name),
-            loading: !props.enableAllAnnotations && areAvailableAnnotationLoading,
-            description: annotation.description,
-            data: annotation,
-            value: annotation.name,
-            displayValue: annotation.displayName,
-        }));
+    const recentAnnotationNames = useSelector(selection.selectors.getRecentAnnotations);
+    const recentAnnotations = recentAnnotationNames.flatMap((name) =>
+        annotations.filter((annotation) => annotation.name === name)
+    );
+
+    const fileNameAnnotation = annotations.filter(
```

Comment:
Doesn't `getSortedAnnotations` include the file name annotation at the beginning of the `annotations` array already?

Reply:
resolved in [0faf718](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/110/commits/0faf718ed4318494cd1d819ddd75348c06910a2c)